rust/sbp/src/messages/invalid.rs

```diff
@@ -0,0 +1,143 @@
+//! Invalid messages. This message type should be created out of two errors:
+//! The SbpMsgParseError, & the CrcError. They correspond to two cases
+//! 1) the Frame is invalid because either it either has an invalid CRC or
+//! it is missing some metadata like msg_type or similar.
+//! 2) The message is invalid because the payload is not large enough and cannot be
+//! parsed into a message. This is the SbpMsgParseError.
+
+use bytes::{Buf, BufMut};
+
+use crate::{de::CrcError, messages::SbpMsgParseError, wire_format::WireFormat, SbpMessage};
+
+/// Invalid messages occur when either the frame or message payload doesn't have enough bytes or
+/// the Crc does not match the messages payload. If the message is well formed with a message
+/// id and crc and payload that all are consistent, but its message type is unknown, it should be
+/// parsed into a Unknown message.
+///
+/// There is something unique about Invalid messages that the payload is actually the entire frame.
+/// This is because it is possible to be able to create invalid messages from invalid frames, that
+/// may not even contain a message payload.
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug, Eq, PartialEq, Clone)]
+pub struct Invalid {
+    /// The message id of the message.
+    #[cfg_attr(
+        feature = "serde",
+        serde(default, skip_serializing_if = "Option::is_none")
+    )]
+    pub msg_id: Option<u16>,
+    /// The message sender_id.
+    #[cfg_attr(
+        feature = "serde",
+        serde(default, skip_serializing_if = "Option::is_none")
+    )]
+    pub sender_id: Option<u16>,
+
+    /// The crc that was in the frame
+    #[cfg_attr(
+        feature = "serde",
+        serde(default, skip_serializing_if = "Option::is_none")
+    )]
+    pub crc: Option<u16>,
+
+    /// payload here is actually the raw frame with the given
+    /// crc, msg_id & sender_id, if any
+    pub payload: Vec<u8>,
+}
+
+impl SbpMessage for Invalid {
+    fn message_name(&self) -> &'static str {
+        "INVALID"
+    }
+
+    fn message_type(&self) -> u16 {
```

Comment:
I think 0xFFFF is a valid message number (MSG_HEARBEAT I think?) -- ideally this iterface would return an `Option` or panic but I'm assuming there's something that makes that sort of change inconvenient?

Reply:
If we're trying to avoid a major version bump of the library I think this sort of change is important enough that it justifies a breaking change (and a major version bump).

Reply:
I don't want to panic here. I think I will just make another breaking change as this pr already does that anyway.

Reply:
👍